From 36bf6ab2b2fced701066c3d3ace2c4a7be428b94 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:49 -0400
Subject: [infiniband] ocrdma: Display proper value for max_mw

Message-id: <7af92170ffade316516d6d63992903714002b6e4.1411571481.git.dledford@redhat.com>
Patchwork-id: 96622
O-Subject: [Patch RHEL7 14/48] RDMA/ocrdma: Display proper value for max_mw
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Selvin Xavier <selvin.xavier@emulex.com>

Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit ac578aef8b9f6095016a12390f0ff638a3a54988)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma.h b/drivers/infiniband/hw/ocrdma/ocrdma.h
index 00b3a29..4ea8496 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma.h
+++ b/drivers/infiniband/hw/ocrdma/ocrdma.h
@@ -67,6 +67,7 @@ struct ocrdma_dev_attr {
  int max_mr;
  u64 max_mr_size;
  u32 max_num_mr_pbl;
+ int max_mw;
  int max_fmr;
  int max_map_per_fmr;
  int max_pages_per_frmr;
diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
index 6eedd2d..f1ec9d9 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
@@ -1016,6 +1016,7 @@ static void ocrdma_get_attr(struct ocrdma_dev *dev,
  attr->local_ca_ack_delay = (rsp->max_pd_ca_ack_delay &
         OCRDMA_MBX_QUERY_CFG_CA_ACK_DELAY_MASK) >>
      OCRDMA_MBX_QUERY_CFG_CA_ACK_DELAY_SHIFT;
+ attr->max_mw = rsp->max_mw;
  attr->max_mr = rsp->max_mr;
  attr->max_mr_size = ~0ull;
  attr->max_fmr = 0;
diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index 40b5559..f8bfdaf 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -89,7 +89,7 @@ int ocrdma_query_device(struct ib_device *ibdev, struct ib_device_attr *attr)
  attr->max_cq = dev->attr.max_cq;
  attr->max_cqe = dev->attr.max_cqe;
  attr->max_mr = dev->attr.max_mr;
- attr->max_mw = 0;
+ attr->max_mw = dev->attr.max_mw;
  attr->max_pd = dev->attr.max_pd;
  attr->atomic_cap = 0;
  attr->max_fmr = 0;
-- 
1.7.1